lib/internal/fs/watchers.js

```diff
@@ -345,11 +357,21 @@ async function* watch(filename, options = kEmptyObject) {
         });
         error.filename = filename;
         handle.close();
-        reject(error);
+        ArrayPrototypePush(queue, error);
+        resolve();
         return;
       }
-
-      resolve({ eventType, filename });
+      if (queue.length < maxQueue) {
+        ArrayPrototypePush(queue, { eventType, filename });
+        resolve();
+      } else if (overflow === 'throw') {
+        const overflowError = genericNodeError('fs.watch maxQueue exceeded');
```

Comment:
This should have a dedicated error code. Can you add one following https://github.com/nodejs/node/blob/ded6b618bed5a9c853f492233ca8a5602b42efe2/doc/contributing/using-internal-errors.md

Reply:
Done in 6720617c7e99cae40b868cb9fcea67dcb7781106